### PR TITLE
fix: remove primary sp id from bucketInfo

### PR DESCRIPTION
**Pull request by Fynnss**
### Description

remove PrimarySpID from BucketInfo

### Rationale

The primary SP ID of each bucket cannot be batch changed when the SP exits. So we should remove it from bucketInfo.

### Example

NA

### Changes

Notable changes: 
* NA

### CODE REVIEW

File: x/challenge/keeper/msg_server_attest.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 
 	"cosmossdk.io/errors"
 	sdkmath "cosmossdk.io/math"
+	types2 "github.com/bnb-chain/greenfield/x/storage/types"
```
Comment (forcodedancing) [inline x/challenge/keeper/msg_server_attest.go:9]: maybe `storagetypes` ?

File: x/challenge/keeper/msg_server_attest.go [hunk 2]
```diff
@@ -65,8 +66,16 @@ func (k msgServer) Attest(goCtx context.Context, msg *types.MsgAttest) (*types.M
...
-	bucketInfo, _ := k.StorageKeeper.GetBucketInfo(ctx, objectInfo.BucketName)
-	if bucketInfo.PrimarySpId != sp.Id {
+	bucketInfo, found := k.StorageKeeper.GetBucketInfo(ctx, objectInfo.BucketName)
+	if !found {
+		return nil, types2.ErrNoSuchBucket.Wrapf("bucket not found when attest")
+	}
+
+	spInState, err := k.StorageKeeper.GetPrimarySPForBucket(ctx, bucketInfo)
+	if err != nil {
+		return nil, errors.Wrapf(err, "get primary for bucket failed. bucketname: %s", bucketInfo.BucketName)
```
Comment (forcodedancing) [inline x/challenge/keeper/msg_server_attest.go:76]: `primary` -> `primary sp`

File: x/storage/keeper/keeper.go [hunk 3]
```diff
@@ -222,15 +221,24 @@ func (k Keeper) doDeleteBucket(ctx sdk.Context, operator sdk.AccAddress, bucketI
...
+		BucketId:                   bucketInfo.Id,
+		GlobalVirtualGroupFamilyId: bucketInfo.GlobalVirtualGroupFamilyId,
 	})
 	return err
 }
 
+func (k Keeper) GetPrimarySPForBucket(ctx sdk.Context, bucketInfo *types.BucketInfo) (*sptypes.StorageProvider, error) {
+	gvgFamily, found := k.virtualGroupKeeper.GetGVGFamily(ctx, bucketInfo.GlobalVirtualGroupFamilyId)
+	if !found {
+		return nil, virtualgroupmoduletypes.ErrGVGFamilyNotExist.Wrapf("gvg family (%d) not found when force delete bucket.", bucketInfo.GlobalVirtualGroupFamilyId)
```
Comment (forcodedancing) [inline x/storage/keeper/keeper.go:236]: The error message is not correct. By the way, it is possible to make it `MustGet...` ? Usually, we had check the existence of bucket.

File: proto/greenfield/virtualgroup/types.proto [hunk 4]
```diff
@@ -35,10 +35,12 @@ message GlobalVirtualGroup {
 message GlobalVirtualGroupFamily {
   // id is the identifier of the global virtual group family.
   uint32 id = 1;
+  // primary_sp_id
+  uint32 primary_sp_id = 2;
```
Comment (krish-nr) [inline proto/greenfield/virtualgroup/types.proto:39]: primary_sp_id should also be added in related events

File: x/challenge/keeper/msg_server_attest.go (hunk 1 above)
Comment (Fynnss) [reply x/challenge/keeper/msg_server_attest.go:9]: Fixed.

File: x/challenge/keeper/msg_server_attest.go (hunk 2 above)
Comment (Fynnss) [reply x/challenge/keeper/msg_server_attest.go:76]: Fixed

File: proto/greenfield/virtualgroup/types.proto (hunk 4 above)
Comment (Fynnss) [reply proto/greenfield/virtualgroup/types.proto:39]: added

File: x/storage/keeper/keeper.go (hunk 3 above)
Comment (Fynnss) [reply x/storage/keeper/keeper.go]: Fixed. `MustGet...` Added.

**Review by forcodedancing — APPROVED** [review]

**Review by krish-nr — APPROVED** [review]
LGTM

**Review by will-2012 — APPROVED** [review]
LGTM.

**Pull request merged**